File: DatingApp.API/DatingApp.DAL/Pagination/UserParams.cs
namespace DatingApp.Services.Pagination
{
    public class UserParams
    {
        // This is the maximum number of elements per page
        private const int MaxPageSize = 50;
        // This is the page number the user will be on, by default it starts at 1
        public int PageNumber { get; set; } = 1;

        // Default page size
        private int _pageSize = 10;
        public int PageSize
        {
            get => _pageSize;
            set => _pageSize = (value > MaxPageSize) ? MaxPageSize : value;
        }

        // This is going to be used for filtering
        // We want to exclude the current user from the filtered results
        public string? CurrentUsername { get; set; }
        // We want to get the desired members with the gender the user wants to see
        public string? Gender { get; set; }
        // This is the min age of the memebers the users want to see
        public int MinAge { get; set; } = 18;
        // Max age
        public int MaxAge { get; set; } = 100;



    }
}

File: Client/src/app/_models/UserParams.ts
// We are using a class here instead of an interface because we want to initalize some default values

import { User } from './User';

// This is helpful for some of the parameters we will pass to the backend
export class UserParams {
  gender: string;
  minAge: number = 18;
  maxAge: number = 99;
  pageNumber: number = 1;
  pageSize: number = 5;
  orderBy: string = 'lastActive';

  constructor(user: User) {
    this.gender = user.gender === 'female' ? 'male' : 'female';
  }
}
